Modules/SubModule_ASL/xASL_wrp_Quantify.m

```diff
@@ -571,3 +577,29 @@ function xASL_wrp_Quantify(x, PWI_Path, pathOutputCBF, M0Path, SliceGradientPath
 	end
 end
 end
+
+function PWI = xASL_wrp_Quantify_MergeSessions(x)
```

Comment:
This still needs to go to a separate function, and it would be `xASL_im_MergePWI4D` right, because that is what we will do? (e.g., it could move elsewhere than quantification).

Can you ensure that all merging stuff is in this function, that would give fewer conflicts with #997 so saves us work.

Reply:
You have several times urged us to never create separate functions that are only called within a single function and that we should always keep them inside the function. 
This is exactly the case of this function. Having it as a separate function will make things more difficult as we will have to check inputs, while here we know the function have a single task.

This will not help with merging at all - because Github can efficiently deal with multiple edits done in the same file as long as they are not editing the same line. 

So moving the function out is
1) Against your previous recommendations.
2) Is not going to make merging with other commits more easy or more difficult.
3) Will make our life more difficult as we will have to work more on this single-purpose function.
4) You have very specifically asked to have this merging here. So it makes no sense arguing now that you want to move it elsewhere.

Reply:
Hopefully you can see this second comment.
I can put this as a separate function, but the motivation shouldn't be merging. Code added at the end of a function will be easily handled by GitHub and will not lead to conflicts. And call of the function is a single line and it will have to be there irrespective if we call internal or separate function.

You would have been absolutely right if the code would have been loosely added to xASL_wrp_QUantify, but it's in a subfunction, so that's not a problem for conflicts. And you always wanted to have one-time-use functions inside and not externally.

Reply:
I see this now indeed, thanks!

1. Because this function can potentially be used at multiple locations (the function should be called xASL_im_MergePWI4D, and allow merging in general, perhaps best called before xASL_wrp_Quantify such that somebody can easily skip it and merge manually, or merge before or after applying M0 in the future etc) I believe this is not something that should be in a subfunction
2. I understand this now, I thought it was completely independent but you do need the merging of parameters in xASL_quant_FSL because there is no separate PWI4D.json yet. I didn't thought of this yet.
3. See (2)
4.  :) -> so my only argument here is that we now have 2 merging functions, one for the image matrix, and one for the metadata TE, PLD, LD. These will become a single function that ignorant of the type of ASL will merge PWI4D and PWI4D.json. So you could consider putting all in the single function that you call from xASL_quant_FSL. So I ask you to consider if merging the merging code in:

1) xASL_module_ASL
2) xASL_wrp_Quantify
3) xASL_quant_FSL

already now all to a single function, that is called from xASL_quant_FSL (in the future from xASL_wrp_Quantify or somewhere else).

That will make it cleaner I believe still (even though I understand now that you need it inside xASL_quant_FSL because we currently don't have a PWI4D.json)